maps/town/strange_tree_interaction.gd

```
@@ -0,0 +1,33 @@
+# Opens up a secret path once the 'Strange Tree' has been interacted with twice.
+extends InteractionTemplateConversation
+
+@onready var _anim: = $AnimationPlayer as AnimationPlayer
+@onready var _popup: = $InteractionPopup as InteractionPopup
+
+
+func _ready() -> void:
+	super._ready()
+	
+	# We want to open the secret path once the player has performed a specific action.
+	# The easiest method is to allow Dialogic to determine when this should happen. This could be
+	# done via the EmitSignal event, the Call event, or, as we've opted to do here, an in-dialogue
+	# signal.
+	# Note that the signal argument must match exactly for the lambda to play.
+	Dialogic.text_signal.connect(
+		func(argument: String):
+			if argument == "clear_secret_path":
```

I'm not managing to get dialogic to run, so I'm not sure what is best to do here; but it seems to me that this is a double opportunity for typos/mistakes that will silently fail.
You could mistype the signal on the side of Dialogic; you could mistype the string array here.

I would recommend one of the following:

1. Runtime check: Force the dialog to always send a signal. Then you can do a match against the argument, and if a signal is not recognized, pop an error (``null` could  be a valid argument, so that'd be a one line change; `assert(argument == null, "noooo")` at the bottom). I don't know if Dialogic always sends an event, if not, this isn't really a guard.
2. Other runtime check: read the global `StrangeTreeExamined` var off Dialogic
3. Static check: have an internal counter increment every time the dialog is triggered (that implies switching the Dialogic dialog resource from code instead of having Dialogic track the number)

---

Agreed with analysis of the issue. It was a clunky solution anyways. The only reason I had done this was to show off in-text Dialogic signals (rather than the events), but the trade-off wasn't worth it here.

It's also apparent to me that Dialogic signal events are pretty common in conversations, so I've added them to the default conversation template (which means a derived class now only needs to implement the _on_dialogic_signal_event callback. It's now already hooked into the interaction when deriving from the template). This has simplified the strange tree interaction and it now follows the same pattern as other interactions.